__tests__/app.test.js

```diff
@@ -77,4 +77,100 @@ describe('GET /api/topics', () => {
             });
         });
     });
+});
+
+describe('GET /api/articles/:article_id', () => {
+    test('response data should be on an "article" key with a 200 http status code', () => {
+        return request(app)
+        .get('/api/articles/1')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body).toHaveProperty('article');
+        });
+    });
+
+    test('each article should have correct object layout', () => {
+        const objectLayout = {
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String)
+        };
+
+        const supertestRequests = [];
+
+        for (let i=1; i<=13; i++) {  // 13 articles in test db
+            const pendingRequest = request(app)
+            .get(`/api/articles/${i}`)
+            .expect(200);
+            
+            const article = new Promise((resolve, reject) => {
+                pendingRequest.then(({ body }) => {
+                    resolve(body.article);
+                })
+                .catch(err => reject(err));
+            });
+
+            supertestRequests.push(article);
+        }
+
+        return Promise.all(supertestRequests)
+        .then(articles => {
+            for (const article in articles) {
+                expect(articles[article]).toMatchObject(objectLayout);
+            }
+        });
```

Comment:
Cool idea to check every item in the database, but possibly a bit of an over test. If it works for one and gives us the response we expect, then we can trust it to work with any id.